### DISCUSSION TITLE
graphing of tire pressures?

### DISCUSSION THREAD

**Discussion by mrbrown8** [teslamate-org/teslamate#1484] [Ideas]
Are tire pressures available through the API? I was thinking it might be interesting how tire pressures might change during driving or as the air temperature changes, or both even.

Is there a status icon for when tire pressure gets too low?

Thanks in advance.

**Comment by LelandSindt** [comment]
Unfortunately, Tire Pressure information is not available from the from the [API](https://tesla-api.timdorr.com/vehicle/state)

**Reply by netkruzer** [reply]
Who do we talk to at Tesla to bring the tire Pressure to the API? It would also be nice to have tire health in the API, since they are now monitoring tire wear in the car.

**Reply by LelandSindt** [reply]
https://twitter.com/elonmusk

**Reply by mortonpd** [reply]
I think this has been added to the API as of 2022.4 but unable to confirm.

**Reply by cwanja** [reply]
It is available via the `vehicle_state` endpoint. 
```
"tpms_pressure_fl":0.0,
"tpms_pressure_fr":0.0,
"tpms_pressure_rl":0.0,
"tpms_pressure_rr":0.0,
```

Although, it seems to only report zeros. Currently my car is parked. I wonder if, just like in-car, it needs time to get the values. But TeslaMate could technically store them and provide you the last value some cadence.

**Reply by djpalmis** [reply]
I hope that tire pressure will be soon shown in drives dashboard or that this data will be correlated with efficiency

**Reply by Aimache** [reply]
According to some testers, the pressure IS available from the API, but reset as soon as the car is set to park. Which means only "non zero values" should be kept/displayed.

**Reply by sunflower-green** [reply]
Aimache:
Great find!

It would be great to have the pressure-information in each drive. But - I think a separate "Tyre pressure" dashboard with all kinds of data/statistics/graphs would be very useful. As we know, tyre pressure does indeed affect consumption - and also tyre wear, so for me it would be a great addon to Teslamate to keep tracking these values over time :-)

**Reply by CarlosCuezva** [reply]
> I hope that tire pressure will be soon shown in drives dashboard or that this data will be correlated with efficiency

I have created a custom panel with the tire pressures in case you are interested in looking at it

![image](https://user-images.githubusercontent.com/5815421/227777272-55ecb929-3bf4-4c46-b871-b8784da31fbb.png)

https://github.com/CarlosCuezva/dashboards-Grafana-Teslamate

**Comment by BriCo84** [comment]
Hope we get this feature :) 

Would like to see my tire pressure changes via graph

**Reply by woyteck1** [reply]
Yes, definitely as I can't see them easily now with new UI :/ 
Too distracting to dig in the menu while driving...

**Comment by Tigwin** [comment]
Tire pressure please. It's available in the API, Tessie is now charting it.

**Reply by djpalmis** [reply]
Teslafi implemented it, too.
But actually I don't see any changes in file versioning..

**Comment by jonferreira** [comment]
would also love to see this being implemented!

**Reply by djpalmis** [reply]
Me too.
I yearn for it.

**Comment by djpalmis** [comment]
Just a gentle reminder

**Comment by NirKli** [comment]
@djpalmis 

I hope this is what you asked for

![image](https://user-images.githubusercontent.com/45952578/174316511-370be3c0-1b96-41ca-8cee-ca9cdcffaa1f.png)

[Added tire pressure #2706](https://github.com/adriankumpf/teslamate/pull/2706)

**Reply by djpalmis** [reply]
So I hope this pull request will be accepted and a new release will be made soon! Thank you anyway!

**Reply by dkropman-Slalom** [reply]
Thank you @NirKli and @djpalmis, I changed docker-compose to edge to get this update. I have slow leak in front tire so this will be very helpful.

**Reply by djpalmis** [reply]
It's all thanks to @NirKli , I did nothing at all. Thank you all guys for this service!

**Reply by metawatchkalle** [reply]
@dkropman-Slalom : Could you please explain how to get the tire-pressure update. What do you mean with "changed docker-compose to edge" Which modification in the YML i have to do?

**Reply by NirKli** [reply]
> @dkropman-Slalom : Could you please explain how to get the tire-pressure update. What do you mean with "changed docker-compose to edge" Which modification in the YML i have to do?

Can you add here your docker-compose file?

**Reply by metawatchkalle** [reply]
I use the standard compose file as described in the wiki:

version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY= #insert a secure key to encrypt your Tesla API tokens
      - DATABASE_USER=teslamate
      - DATABASE_PASS= #insert your secure database password!
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
    ports:
      - 4000:4000
    volumes:
      - ./import:/opt/app/import
    cap_drop:
      - all

  database:
    image: postgres:14
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD= #insert your secure database password!
      - POSTGRES_DB=teslamate
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS= #insert your secure database password!
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3000:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    # ports:
    #   - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:

**Reply by NirKli** [reply]
> I use the standard compose file as described in the wiki:
> 
> 
> 
> version: "3"
> 
> 
> 
> services:
> 
>   teslamate:
> 
>     image: teslamate/teslamate:latest
> 
>     restart: always
> 
>     environment:
> 
>       - ENCRYPTION_KEY= #insert a secure key to encrypt your Tesla API tokens
> 
>       - DATABASE_USER=teslamate
> 
>       - DATABASE_PASS= #insert your secure database password!
> 
>       - DATABASE_NAME=teslamate
> 
>       - DATABASE_HOST=database
> 
>       - MQTT_HOST=mosquitto
> 
>     ports:
> 
>       - 4000:4000
> 
>     volumes:
> 
>       - ./import:/opt/app/import
> 
>     cap_drop:
> 
>       - all
> 
> 
> 
>   database:
> 
>     image: postgres:14
> 
>     restart: always
> 
>     environment:
> 
>       - POSTGRES_USER=teslamate
> 
>       - POSTGRES_PASSWORD= #insert your secure database password!
> 
>       - POSTGRES_DB=teslamate
> 
>     volumes:
> 
>       - teslamate-db:/var/lib/postgresql/data
> 
> 
> 
>   grafana:
> 
>     image: teslamate/grafana:latest
> 
>     restart: always
> 
>     environment:
> 
>       - DATABASE_USER=teslamate
> 
>       - DATABASE_PASS= #insert your secure database password!
> 
>       - DATABASE_NAME=teslamate
> 
>       - DATABASE_HOST=database
> 
>     ports:
> 
>       - 3000:3000
> 
>     volumes:
> 
>       - teslamate-grafana-data:/var/lib/grafana
> 
> 
> 
>   mosquitto:
> 
>     image: eclipse-mosquitto:2
> 
>     restart: always
> 
>     command: mosquitto -c /mosquitto-no-auth.conf
> 
>     # ports:
> 
>     #   - 1883:1883
> 
>     volumes:
> 
>       - mosquitto-conf:/mosquitto/config
> 
>       - mosquitto-data:/mosquitto/data
> 
> 
> 
> volumes:
> 
>   teslamate-db:
> 
>   teslamate-grafana-data:
> 
>   mosquitto-conf:
> 
>   mosquitto-data:

You have to change 'teslamate/teslamate:latest' to 'teslamate/teslamate:edge', and 'teslamate/grafana:latest' to teslamate/grafana:edge' if you want to use the latest development version

**Reply by metawatchkalle** [reply]
Works! Thank you very much. After some googling I could have figured that out by myself.....

**Reply by Tigwin** [reply]
How come my tire pressure is in bar instead of psi? It's showing 2.7 bar, which I then have to use a calculator to figure out the psi

**Reply by NirKli** [reply]
> How come my tire pressure is in bar instead of psi? It's showing 2.7 bar, which I then have to use a calculator to figure out the psi

You can change it on Tire Pressure Units button

![image](https://user-images.githubusercontent.com/45952578/176476205-a56027b4-68fe-4cad-b9be-880eb3a2f7c0.png)

**Reply by Aimache** [reply]
It's a little confusing this unit selector is not part of the other unit selections in the vehicule settings (with temperature/distance), but it helps for sure!

**Reply by cwanja** [reply]
> > How come my tire pressure is in bar instead of psi? It's showing 2.7 bar, which I then have to use a calculator to figure out the psi
> 
> You can change it on Tire Pressure Units button
> 
> ![image](https://user-images.githubusercontent.com/45952578/176476205-a56027b4-68fe-4cad-b9be-880eb3a2f7c0.png)



> It's a little confusing this unit selector is not part of the other unit selections in the vehicule settings (with temperature/distance), but it helps for sure!

Is it a set it once and it retains the setting? Or do you have to set it each time you open the dashboard? 

One option is at moving the dropdown to a setting (like @Aimache mentioned) that writes to the database as a configuration and then Grafana references the setting as a variable to drive the dashboard setting.

**Reply by NirKli** [reply]
> It's a little confusing this unit selector is not part of the other unit selections in the vehicule settings (with temperature/distance), but it helps for sure!

Ummm, actually its good idea, didn't thought about it

**Reply by cwanja** [reply]
@NirKli #2727 is going. But it is only adding the selector to the settings page. Will require writing that setting to the database, adding the variable to grafana to use the setting and then updating the drive details to reference the variable and not use the dropdown. Just a starting point.

**Comment by woyteck1** [comment]
Will this be in the new release at some point? Thanks

    On Wednesday, 29 June 2022 at 16:43:43 BST, cwanja ***@***.***> wrote:  
 
 


@NirKli #2727 is going. But it is only adding the selector to the settings page. Will require writing that setting to the database, adding the variable to grafana to use the setting and then updating the drive details to reference the variable and not use the dropdown. Just a starting point.

**Reply by cwanja** [reply]
Assuming it gets approved and the additional work is completed to make it functional, yes.

**Comment by 3rdStng** [comment]
Pretty sure the tire pressure has been added into Teslamate. I see the dashboard, but there is zero data in any of my drives. Am I incorrect that this has been committed? Or any idea why my dashboard is blank?

![image](https://user-images.githubusercontent.com/24888063/227663822-2ff3e1a3-0e6a-4e87-82f6-7ec6ce570ea5.png)

**Comment by netkruzer** [comment]
My tire pressure shows. Have you tried restarting teslamate or pulling the latest version?

![image](https://user-images.githubusercontent.com/7841302/227669607-5424304a-8e86-42da-9d90-fbf49f109bf7.png)




I'm on Version 1.27.2

**Reply by 3rdStng** [reply]
Yes and yes. Also running 1.27.2. I don't know enough about sql to know the command to query if anything is being written to the Db. Also don't know if the year of the car matters. I have a 2019 M3 running 2022.40.30.10.

**Reply by netkruzer** [reply]
I have an early 2019 M3 (AP2.5), running 2023.2.12 and a 2022 MY, tire pressures show up fine on both. I wonder if you need to sign into Tesla again to resync.

**Reply by 3rdStng** [reply]
I hadn't signed out in quite a while. Before Tesla changed the auth token requirements. Signed out, got my new tokens, and successfully signed in. How long until I need to refresh the tokens again? I've played with the Tesla integration into Home Assistant, but the manual refresh was way too frequent for me. I'm assuming the refresh token is what is supposed to take care of this.

This looks like it might have done it. Thank you @netkruzer.

**Comment by 3rdStng** [comment]
Not sure where to start investigating this one. Signing out and back in enabled me to see the tire pressure readings when the car is stopped. However, while in drive, there is no TP data and such, nothing is recorded in my drive details too. 

Here are captures while the car is being driven. Should there be data?
![image](https://user-images.githubusercontent.com/24888063/231312189-62f770dd-493b-409d-bd6a-c11478028344.png)
![image](https://user-images.githubusercontent.com/24888063/231312353-39fbe018-4deb-462d-823e-0cae87c23ef0.png)

**Reply by NirKli** [reply]
> Not sure where to start investigating this one. Signing out and back in enabled me to see the tire pressure readings when the car is stopped. However, while in drive, there is no TP data and such, nothing is recorded in my drive details too.
> 
> Here are captures while the car is being driven. Should there be data? ![image](https://user-images.githubusercontent.com/24888063/231312189-62f770dd-493b-409d-bd6a-c11478028344.png) ![image](https://user-images.githubusercontent.com/24888063/231312353-39fbe018-4deb-462d-823e-0cae87c23ef0.png)

Do you see the tire pressure on the Tesla app?

<img width="346" alt="image" src="https://user-images.githubusercontent.com/45952578/231355826-31e9a060-bba3-4025-ae5d-c1e59270cf2c.png">

**Reply by CarlosCuezva** [reply]
The only condition to display the data on these panels is that the left front sensor has data.

**Reply by NirKli** [reply]
> The only condition to display the data on these panels is that the left front sensor has data.

Maybe that is the issue, Interesting

**Comment by 3rdStng** [comment]
Yes. I can see the PSI within the app. When the car is parked I can see the PSI in both panels of the TP dashboard. But while the car is driving, there is no TP displayed in TeslaMate/Grafana. The below screenshot is while the car is driving. 

I'm curious if this is why it's not being recorded for the drives? Or is there something wrong with my DB not collecting/storing the data? I don't know where to start troubleshoot this either.
![Screenshot_20230411-222552](https://user-images.githubusercontent.com/24888063/231359619-16b857ac-5ef3-44a6-b01d-3ab700fbf0fd.png)

**Reply by NirKli** [reply]
> Yes. I can see the PSI within the app. When the car is parked I can see the PSI in both panels. But while the car is driving, there is no TP displayed. I'm curious if this is why it's not being recorded for the drives? Or is there something wrong with my DB not collecting/storing the data? ![Screenshot_20230411-222552](https://user-images.githubusercontent.com/24888063/231359619-16b857ac-5ef3-44a6-b01d-3ab700fbf0fd.png)

Can you edit the tire pressure query by hover your mouse on the tire pressure dashboard and click e and then delete the ` AND
    tpms_pressure_fl is not null` statement and click Table view on the top.
Post here a picture of the data

<img width="1055" alt="image" src="https://user-images.githubusercontent.com/45952578/231364293-1555720a-8ed4-4fef-80bf-dc61a2ee953d.png">

**Comment by 3rdStng** [comment]
I removed that line and my table has a log of holes in it afterwards. 
![image](https://user-images.githubusercontent.com/24888063/231513789-e192b44a-1f42-4333-9ad2-3fc8f178aed3.png)

I had to scroll down to see some data. Everything with yesterday's date is blank.
![image](https://user-images.githubusercontent.com/24888063/231514355-cf0f2bd8-cd27-4c8a-b892-34142e4f1c88.png)

Which confuses me more as the TP dashboard without any edits shows data in the table.
![image](https://user-images.githubusercontent.com/24888063/231514933-b7389716-50b4-4df0-a75f-6dd86e1e972e.png)

**Reply by NirKli** [reply]
Try this query and post the data,

```sql
SELECT
    $__time(date),
    convert_tire_pressure(tpms_pressure_fl,'$pressure_unit') AS tpms_pressure_front_left_$pressure_unit,
    convert_tire_pressure(tpms_pressure_fr,'$pressure_unit') AS tpms_pressure_front_right_$pressure_unit,
    convert_tire_pressure(tpms_pressure_rl,'$pressure_unit') AS tpms_pressure_rear_left_$pressure_unit,
    convert_tire_pressure(tpms_pressure_rr,'$pressure_unit') AS tpms_pressure_rear_right_$pressure_unit
FROM
    positions
WHERE
    car_id = $car_id AND
    $__timeFilter(date)
ORDER BY
    date ASC
```

**Reply by 3rdStng** [reply]
Do I run this at the console from the psql prompt? Or change the query within the TP panel? I replaced the SELECT FROM WHERE section of the panel and left the rest there. The results an error about no drive_id. Running it from the console immediately returned back to the prompt with no output.

**Reply by NirKli** [reply]
> Do I run this at the console from the psql prompt? Or change the query within the TP panel? I replaced the SELECT FROM WHERE section of the panel and left the rest there. The results an error about no drive_id. Running it from the console immediately returned back to the prompt with no output.

```SQL
SELECT
	date,
    tpms_pressure_fl AS tpms_pressure_front_left,
    tpms_pressure_fr AS tpms_pressure_front_right,
    tpms_pressure_rl AS tpms_pressure_rear_left,
    tpms_pressure_rr AS tpms_pressure_rear_right
FROM
    positions
WHERE
    car_id = 1
ORDER BY
    date DESC
Limit 1000
```
This query is more generic, you can run it everywhere.
I'm using pgAdmin to query teslamate postgres

**Reply by 3rdStng** [reply]
Am I doing something wrong? Or do I have no data? I am not a sql expert by any means either. 
![image](https://user-images.githubusercontent.com/24888063/232244985-65b60340-dc49-4edb-968f-9727b2a7ba50.png)

**Reply by NirKli** [reply]
> Am I doing something wrong? Or do I have no data? I am not a sql expert by any means either. ![image](https://user-images.githubusercontent.com/24888063/232244985-65b60340-dc49-4edb-968f-9727b2a7ba50.png)

My bad, forgot the `;` in the end of the statement

```sql
SELECT date, tpms_pressure_fl AS tpms_pressure_front_left, tpms_pressure_fr AS tpms_pressure_front_right, tpms_pressure_rl AS tpms_pressure_rear_left, tpms_pressure_rr AS tpms_pressure_rear_right FROM positions WHERE car_id = 1 ORDER BY date DESC Limit 1000;
```

**Reply by 3rdStng** [reply]
That yielded data.
[output.txt](https://github.com/adriankumpf/teslamate/files/11240324/output.txt)

**Reply by NirKli** [reply]
> That yielded data. [output.txt](https://github.com/adriankumpf/teslamate/files/11240324/output.txt)

I think there's an issue with grafana, the data looks fine.
Did you made any changes to grafana that came from teslamate?

**Reply by 3rdStng** [reply]
No changes. Other than I didn't use the docker compose file and pulled my own dockers. I am not running the latest Grafana docker though. I am running grafana/grafana:8.5.15. I usually watch the Teslamate release notes for when Grafana gets a bump. But haven't seen anything recently. I just looked at the docker compose file and see the image is teslamate/grafana:latest. According to the dockerfile in the Grafana directory, it looks like this just pulls the same image I have. 
`FROM grafana/grafana:8.5.15`

I also re-imported the drives dashboard again, just in case. No change though.

**Comment by CarlosCuezva** [comment]
The Tesla API does not always send the values of the tire pressures, so you always have to filter for null values

**Comment by 3rdStng** [comment]
Thank you to @OrangeBean for the drive-details dashboard updates. The missing dataSource for pressure units resolved my inability to see my tire pressure in the dashboards.
